### PR TITLE
community: update for compatibility with latest Meilisearch version

### PR DESCRIPTION
**Pull request by CaroFG**
Thank you for contributing to LangChain!

- [x] **PR title**: "package: description"
  - Where "package" is whichever of langchain, community, core, experimental, etc. is being modified. Use "docs: ..." for purely docs changes, "templates: ..." for template changes, "infra: ..." for CI changes.
  - Example: "community: add foobar LLM"

- [x] **PR message**:
    - **Description:** Updates Meilisearch vectorstore for compatibility with v1.6 and above. Adds embedders settings and embedder_name which are now required.

- [x] **Add tests and docs**: If you're adding a new integration, please include
  1. a test for the integration, preferably unit tests that do not rely on network access,
  2. an example notebook showing its use. It lives in `docs/docs/integrations` directory.

- [x] **Lint and test**: Run `make format`, `make lint` and `make test` from the root of the package(s) you've modified. See contribution guidelines for more: https://python.langchain.com/docs/contributing/ 

Additional guidelines:
- Make sure optional dependencies are imported within a function.
- Please do not add dependencies to pyproject.toml files (even optional ones) unless they are required for unit tests.
- Most PRs should not touch more than one package.
- Changes should be backwards compatible.
- If you are adding something to community, do not re-import it in langchain.

If no one reviews your PR within a few days, please @-mention one of baskaryan, efriis, eyurtsev, hwchase17.

### CODE REVIEW

**Comment by CaroFG** [comment]
Hello! I don't want to be a bother, but since it's mentioned in the PR template, I'm tagging @efriis to make sure you don't forget this PR :D

File: libs/community/langchain_community/vectorstores/meilisearch.py [hunk 1]
```diff
@@ -67,6 +67,7 @@ class Meilisearch(VectorStore):
             embeddings = OpenAIEmbeddings()
             vectorstore = Meilisearch(
                 embedding=embeddings,
+                embedders=embedders,
```
Comment (baskaryan) [inline libs/community/langchain_community/vectorstores/meilisearch.py:70]: what are embedders? could we show how you'd define embedders in this example?

Comment (CaroFG) [reply libs/community/langchain_community/vectorstores/meilisearch.py:70]: In Meilisearch versions greater than 1.6, to use vector search, you must first configure the embedders index setting. This setting is a dictionary structure in which each key corresponds to an embedder. 

The configuration options within an embedder object vary depending on the value assigned to its source field, which specifies the third-party service used for generating embeddings. In this context, the source is `userProvided`.

For `userProvided` sources, it's required to include a dimensions field, which outlines the size of the vectors.

An example configuration for embedders could look like this:

```
embedders = {
    "theEmbedderName": {
        "source": "userProvided",
        "dimensions": "1536"
    }
}
```

The use can define multiple embedders per index to accommodate more demanding use cases. For example, a user can define one embedder to generate text from a document image, and another embedder to translate the original document text fields to let users speaking other languages search products in their native language.

File: libs/community/langchain_community/vectorstores/meilisearch.py [hunk 2]
```diff
@@ -125,10 +134,11 @@ def add_texts(
             metadata = metadatas[i]
             metadata[self._text_key] = text
             embedding = embedding_vectors[i]
+            embedder_name
```
Comment (baskaryan) [inline libs/community/langchain_community/vectorstores/meilisearch.py:137]: stray line?

File: libs/community/langchain_community/vectorstores/meilisearch.py (hunk 1 above)
Comment (baskaryan) [reply libs/community/langchain_community/vectorstores/meilisearch.py:70]: gotcha! should we that snippet in the docstring example?

**Review by baskaryan — APPROVED** [review]

**Pull request merged**